### PR TITLE
(Feature) | Simplify XMLNode API to prevent unexpected behavior

### PR DESCRIPTION
**Pull request by eneko**
- [x] I've read, understood, and done my best to follow the [*CONTRIBUTING guidelines*](https://github.com/mindbody/conduit/blob/master/CONTRIBUTING.md).

This pull request includes (pick all that apply):

- [ ] Bugfixes
- [ ] New features
- [x] Breaking changes
- [x] Documentation updates
- [x] Unit tests
- [ ] Other

### Summary
`XMLNode` is very powerful, allowing for very easy retrieval of values from nested XML nodes. This is achieved by defaulting to `breadthFirst` XML tree traversal.

However, this feature requires developers of consuming applications and frameworks to pay extra attention on the actual XML structure and what is happening behind the scenes when the value is retrieved.

In other words, the behavior of methods like `getValue()` can be unexpected if the developer is not aware of the existence of false-positive matches in the value retrieval.

### Implementation
- Default values for `nodes(named:traversal:)` and `node()` methods have been removed and traversal algorithm must be now set explicitly.
- `getValue(name:)` has been updated to always use `.firstLevel` only.
- New method `findValue(name:traversal:)` has been added, and requires the traversal algorithm to be set explicitly.

### Test Plan
- Run unit and integration tests with `rake test`
- Approval by CI

### CODE REVIEW

**Comment by rminus** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Source files were modified – don’t forget to update the CHANGELOG if there are code changes.<br />
You can find examples within the <a href="https://github.com/mindbody/Conduit/blob/master/CHANGELOG.md">CHANGELOG.md</a></td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>
